### PR TITLE
add delta load convenience step

### PR DESCRIPTION
**Pull request by tanonev**
PTAL @sb2nov

### CODE REVIEW

File: dataduct/steps/delta_load.py [hunk 1]
```diff
@@ -0,0 +1,228 @@
...
+            source_relation = Table(SqlScript(filename=parse_path(source)))
+        else:
+            source_relation = SelectStatement(
+                SqlScript(sql=sql, filename=parse_path(script)).sql())
+
+        delta_clause = """
+            WHERE {date_column} >=
+                COALESCE(
+                    (SELECT MAX({date_column}) FROM {production}),
+                    '2000-01-01'::DATE
```
Comment (sb2nov) [inline dataduct/steps/delta_load.py]: change this as it might not be safe for other companies

File: dataduct/steps/delta_load.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+            depends_on=depends_on
+        )
+        return upsert_pipeline_object
+
+    @classmethod
+    def get_table_from_def(cls, table_definition):
+        with open(parse_path(table_definition)) as f:
+            table_def_string = f.read()
+
+        table = Table(SqlStatement(table_def_string))
```
Comment (sb2nov) [inline dataduct/steps/delta_load.py]: sql_script has a filename param in the constructor so you can use that.

**Comment by sb2nov** [comment]
Chatted offline and still WIP

**Comment by sb2nov** [comment]
LGTM

**Comment by sb2nov** [comment]
Actually how is this better than upsert with custom sql. or is the intention just to make it a bit more convenient ?

**Comment by tanonev** [comment]
This is meant to be a convenience method; the idea is if your upsert
statement fits the delta load pattern already, you can switch it from an
upsert to a delta load and get a performance benefit (and lack of
dependency on full history) for free.

**Pull request merged**